### PR TITLE
Cleanup: Make all ASTs lazy (no _AbstractCodeBuilder)

### PR DESCRIPTION
**Pull request by matanlurey**
Closes https://github.com/dart-lang/code_builder/issues/6

### CODE REVIEW

**Comment by matanlurey** [comment]
This also makes the package `pub publish`-able (cc @kevmoo)

File: README.md [hunk 1]
```diff
@@ -3,20 +3,35 @@
...
-Code builder also adds a more narrow and user-friendly API. For example
-creating a class with a method is an easy affair:
+## Experimental
+
+While `code_builder` is considered *stable*, the APIs are subject to
+frequent breaking change - a number of Dart language features are not
+yet implemented that make it unsuitable for all forms of code
+generation. 
+
+**Contributions are [welcome][welcome]!**
```
Comment (yjbanov) [inline README.md]: Should the second "welcome" be inside round parens: (welcome)?

File: lib/src/builders/class_builder.dart [hunk 2]
```diff
@@ -5,6 +5,8 @@
 part of code_builder;
 
 /// Builds a [ClassDeclaration] AST.
+///
+///
```
Comment (yjbanov) [inline lib/src/builders/class_builder.dart]: I'm on the edge of my seat in suspense, waiting for docs to appear ;)

File: lib/src/builders/file_builder.dart [hunk 3]
```diff
@@ -36,33 +40,41 @@ class ExportBuilder extends _AbstractCodeBuilder<ExportDirective> {
...
-  factory ImportBuilder(String uri, {String as}) {
-    var astNode = _createImportDirective()..uri = _stringLiteral("'$uri'");
-    if (as != null) {
-      astNode
-        ..asKeyword = _as
-        ..prefix = _stringIdentifier(as);
-    }
-    return new ImportBuilder._(astNode);
-  }
+  const factory ImportBuilder(String uri, {String as}) = ImportBuilder._;
```
Comment (yjbanov) [inline lib/src/builders/file_builder.dart]: `as` is a Dart keyword. How about `withPrefix` or just `prefix`? That's what the language spec calls them I think.

File: README.md (hunk 1 above)
Comment (matanlurey) [inline README.md]: `[foo][bar]` means that `[bar]` is defined lower in the markdown document, i.e:

`[bar]: http://some-long-link-that-is-awkward-inline.com`

File: lib/src/builders/class_builder.dart (hunk 2 above)
Comment (matanlurey) [inline lib/src/builders/class_builder.dart]: Reverted

File: lib/src/builders/file_builder.dart (hunk 3 above)
Comment (matanlurey) [inline lib/src/builders/file_builder.dart]: Done. Changed to `prefix`

**Comment by matanlurey** [comment]
Assuming travis passes (does locally) I'll squash and merge. Thanks!

File: README.md (hunk 1 above)
Comment (yjbanov) [inline README.md]: Does this mean that there's an error in the `[analyzer](analyzer)` link above?

**Pull request merged**